connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaStatusBackingStore.java

```diff
@@ -448,14 +448,14 @@ public boolean isDeleted() {
             return deleted;
         }
 
-        public boolean canWrite(T status) {
-            return value != null &&
```

Comment:
The motivation for this change is that connector/task status listeners' `onFailure()` callbacks invoke `StatusBackingListener.putSafe()`, but the status was not being reflected if there wasn't a previous value (e.g. STARTED, which is written via the `StatusBackingListener.put()` path) since `canWrite()` was returning false.